### PR TITLE
Recipe updates for v3.4.1

### PR DESCRIPTION
**Pull request by trexfeathers**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] ~~Bumped the build number (if the version is unchanged)~~
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by trexfeathers** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ outputs:
         - cftime >=1.5
         - dask-core >=2.26
         - matplotlib-base >=3.5
-        - netcdf4 <1.6.1
+        - netcdf4
```
Comment (valeriupredoi) [inline recipe/meta.yaml:31]: yay :partying_face:

**Comment by trexfeathers** [comment]
@valeriupredoi does conda-forge/conda-forge-repodata-patches-feedstock#358 mean anything for this PR?

**Review by valeriupredoi — COMMENTED** [review]
@trexfeathers great to see this going :tada: Did you also add python=3.11 as supported by the new iris?

**Comment by trexfeathers** [comment]
>Did you also add python=3.11 as supported by the new iris?

Not explicitly - that would go beyond the minimal patch we are aiming for. We won't stop you trying it, but can't guarantee it will work.

**Comment by valeriupredoi** [comment]
> @valeriupredoi does [conda-forge/conda-forge-repodata-patches-feedstock#358](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/358) mean anything for this release? I'm assuming not?

not for this one it doesn't - the patch stops at iris=3.4.0 and I think it's best you guys keep it that way unless you want to re-release 3.4.0 with the new segfault slayer that this version contains

**Comment by valeriupredoi** [comment]
> > Did you also add python=3.11 as supported by the new iris?
> 
> Not explicitly - that would go beyond the minimal patch we are aiming for. We won't stop you trying Python 3.11, but can't guarantee it will work.

excellent! We'll plug 3.11 only when iris-esmf-regrid will support it (I think it's still 3.10-limited atm)

**Review by stephenworsley — APPROVED** [review]
Looks good.

**Pull request merged**